### PR TITLE
Update ERC-7092: Move to Review

### PR DESCRIPTION
**Pull request by Edoumou**
The move from EIPs to ERcs is causing conflicts due to folder name changed for all eips.

- PR Closed -> https://github.com/ethereum/EIPs/pull/7667

Opening this new PR

When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: ERCS/erc-7092.md [hunk 1]
```diff
@@ -1,59 +1,35 @@
...
-3. Increased liquidity: Fractioanal ownership will bring new investors in the bond market, this will increase liquidity in the bond market.
-4. Cost savings: By replacing intermediaries with smart contracts, bond's tokenization will reduce costs associated with the bond issuance and management.
-5. Easy accessibility and 24/7 trading: Tokenized bonds are supposed to be traded on digital platforms such as decentralized exchanges. Therefore, they will be more accessible compared to tradional bond market.
+Fixed-income instruments are a widely utilized asset class for corporations and other entities raising funds. However, transitioning to tokenized bonds is challenging due to existing standards like [ERC-3475](./erc-3475.md), which introduces unfamiliar concepts and leads to unnecessary gas consumption. Additionally, the lack of named variables like coupon, maturity date, and principal, makes it difficult to implement ERC-3475 since developers need to remember which metadata is assigned to each parameter.
```
Comment (lightclient) [inline ERCS/erc-7092.md:19]:
```suggestion
Fixed-income instruments are a widely utilized asset class for corporations and other entities raising funds. However, transitioning to tokenized bonds is challenging due to existing standards like [ERC-3475](./eip-3475.md), which introduces unfamiliar concepts and leads to unnecessary gas consumption. Additionally, the lack of named variables like coupon, maturity date, and principal, makes it difficult to implement ERC-3475 since developers need to remember which metadata is assigned to each parameter.
```

**Review by lightclient — CHANGES_REQUESTED** [review]

**Review by lightclient — APPROVED** [review]

File: ERCS/erc-7092.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ This proposal introduces fixed-income financial bonds with key characteristics d
 
 ## Motivation
 
-Fixed-income instruments are a widely utilized asset class for corporations and other entities raising funds. However, transitioning to tokenized bonds is challenging due to existing standards like [ERC-3475](./erc-3475.md), which introduces unfamiliar concepts and leads to unnecessary gas consumption. Additionally, the lack of named variables like coupon, maturity date, and principal, makes it difficult to implement ERC-3475 since developers need to remember which metadata is assigned to each parameter.
+Fixed-income instruments are a widely utilized asset class for corporations and other entities raising funds. However, transitioning to tokenized bonds is challenging due to existing standards like [ERC-3475](./eip-3475.md), which introduces unfamiliar concepts and leads to unnecessary gas consumption. Additionally, the lack of named variables like coupon, maturity date, and principal, makes it difficult to implement ERC-3475 since developers need to remember which metadata is assigned to each parameter.
```
Comment (Edoumou) [inline ERCS/erc-7092.md:19]: Hello @lightclient,

All folders and files have been renamed _erc-Number_ in the ERCs repository, therefore referencing with the path **_./eip-3475.md_** leads to an error, that's the case now. You should revert the change to **_./erc-3475.md_**.

Let me know if I am wrong.

File: ERCS/erc-7092.md [hunk 3]
```diff
@@ -403,9 +508,15 @@ contract ERC7092Convertible is ERC7092 {
 }
 ```
 
+### Identity Registry
+
+This standard is designed specifically for tokenizing bonds. It does not inherently manage information pertaining to bondholders' identities. However, to enhance compliance with regulatory requirements and improve transparency, it is strongly RECOMMENDED to implement an identity registry to store the identity addresses of all authorized investors.
```
Comment (SamWilsn) [inline ERCS/erc-7092.md:513]: Similarly, no requirements in reference implementation.

**Review by SamWilsn — COMMENTED** [review]
Couple really minor nits, and then this should be good for review.

File: ERCS/erc-7092.md [hunk 4]
```diff
@@ -68,259 +44,387 @@ interface IERC7092 {
...
-the `denomination`, and the `maturity date`. Those information are very important for bondholders to know the return on the capital invested.
+The use of traditional finance terminology, like `issueVolume` and `principalOf`, is aimed at maintaining consistency with traditional bond language, which eases the adaptation for traditional entities.
 
-### Total Supply
+### Total Supply and Account Balance
 
-We made the choice not to define the `totalSupply` function beacause it can be derived from both the `issue volume` and the `denomination`.
-However it is RECOMMENDED to define the `totalSupply` function in any contract that implements this standard. In that case,
-the `total supply` MUST be equal to the ratio of the `issue volume` and the `denomination`.
+The `totalSupply` and `balanceOf` functions are not defined as they can be derived from `issueVolume` and `principalOf`, and `denomination`. However, it's RECOMMENDED to define these functions in any contract implementing this standard, ensuring the proper relationship between these values.
```
[the remark this answers is not in the document]
Comment (Edoumou) [reply ERCS/erc-7092.md]: Thank you @SamWilsn .

Just fixed it.

File: ERCS/erc-7092.md (hunk 3 above)
Comment (Edoumou) [reply ERCS/erc-7092.md]: > Similarly, no requirements in reference implementation.

Thank you @SamWilsn .

I just fixed it.

File: ERCS/erc-7092.md [hunk 5]
```diff
@@ -1,326 +1,156 @@
...
 
-    /**
-    * @notice Moves all bonds to address `_to`
-    * @param _to the address to send the bonds to
-    * @param _data additional information provided by the token holder
-    *
-    * OPTIONAL - interfaces and other contracts MUST NOT expect these values to be present. The method is used to improve usability.
-    */
-    function transferAll(address _to, bytes calldata _data) external returns(bool);
+A detailed description of the functions can be found [here](../assets/erc-7092/IERC7092Batch.sol)
```
Comment (SamWilsn) [inline ERCS/erc-7092.md:80]: The specification section should be understandable on its own, without having to refer to the reference implementation. Would you mind duplicating/moving the function descriptions here?

File: ERCS/erc-7092.md [hunk 6]
```diff
@@ -1,326 +1,156 @@
...
+### Additional bond parameters Interface
 
-    /**
-    * @notice Remove the allowance for `_spender`
-    * @param _spender the address to remove the authorization by from
-    *
-    * OPTIONAL - interfaces and other contracts MUST NOT expect these values to be present. The method is used to improve usability.
-    */
-    function decreaseAllowanceForAll(address _spender) external returns(bool);
+Functions in the `IERC7092ESG` interface are OPTIONAL. This interface MAY be used to improve the standard usability.
```
Comment (SamWilsn) [inline ERCS/erc-7092.md:73]: This is a tad ambiguous. Is the whole interface optional, or are the individual functions within it optional? The former is fine, but the latter will mess up the ERC-165 interface id.

If the whole interface is optional, I'd recommend:

```suggestion
The `IERC7092ESG` interface is OPTIONAL for contracts implementing this proposal. This interface MAY be used to improve the standard usability.
```

On the other hand, if the individual functions are optional, you could move each one into its own interface.

File: ERCS/erc-7092.md [hunk 7]
```diff
@@ -1,326 +1,156 @@
...
 
 ## Backwards Compatibility
 
-Beacause some functions like `totalSupply` or `balanceOf` are not implemented, the ERC-7092 SHOULD NOT extend existing standards,
-like [ERC-20](./eip-20) or [ERC-1155](./eip-1155). The ERC-7092 is a representation of a new token for bonds with all bonds characteristics
-and functionalities already attached to it.
-
-**_For the reasons mentionned above, we recommand a pure implementation of the standard_** to issue tokenized bonds, since any hybrid solution
-with other standards mentionned above SHOULD fail.
+This ERC SHOULD NOT extend existing standards like [ERC-20](./erc-20.md) or [ERC-1155](./erc-1155.md) due to the absence of certain functions like `totalSupply` or `balanceOf`. A pure implementation of this standard is RECOMMENDED for issuing tokenized bonds, as any hybrid solution with other mentioned standards SHOULD fail.
```
Comment (SamWilsn) [inline ERCS/erc-7092.md:146]:
```suggestion
This ERC is not backwards compatible with existing standards like [ERC-20](./erc-20.md) or [ERC-1155](./erc-1155.md) due to the absence of certain functions like `totalSupply` or `balanceOf`. A pure implementation of this standard is RECOMMENDED for issuing tokenized bonds, as any hybrid solution with other mentioned standards SHOULD fail.
```

File: ERCS/erc-7092.md [hunk 8]
```diff
@@ -1,326 +1,156 @@
...
-
-Some bonds have embedded options attached to them. As an example we can cite:
-
-1. Callable bonds that have an option that gives the issuer the right to retire bonds before they mature.
-2. Puttable bonds that have an option that gives investors the right to retire bonds before they mature.
-3. Convertible bonds that gives investors the right to convert their bonds to equity.
+The complete Reference Implementation can be found [here](../assets/erc-7092/ERC7092.sol).
 
-Bonds with embedded options can be created by inheriting from the basic ERC-7092 that integrates the proposed interface.
+Bonds with embedded options like callable, puttable, or convertible bonds can be created by inheriting from the reference [ERC-7092](../assets/erc-7092/ERC7092.sol) that integrates the proposed interface.
```
Comment (SamWilsn) [inline ERCS/erc-7092.md:153]:
```suggestion
Bonds with embedded options like callable, puttable, or convertible bonds can be created by inheriting from the reference [`ERC7092.sol`](../assets/erc-7092/ERC7092.sol) that integrates the proposed interface.
```

File: ERCS/erc-7092.md (hunk 5 above)
Comment (Edoumou) [reply ERCS/erc-7092.md:80]: > The specification section should be understandable on its own, without having to refer to the reference implementation. Would you mind duplicating/moving the function descriptions here?

Hello @SamWilsn,

All functions that manage the entire balance of an account have been removed for security purposes. Therefore the`transferferAll` function is not part of the standard anymore. This will be update when merging the PR to Review.

File: ERCS/erc-7092.md [hunk 9]
```diff
@@ -165,162 +135,289 @@ interface IERC7092 {
...
-    function transferAllFrom(address _from, address _to, bytes calldata _data) external returns(bool);
+    function decimals() external view returns(uint8);
 
     /**
-    * @notice MUST be emitted when bonds are transferred
-    * @param _from the account that owns bonds
-    * @param _to the account that receives the bond
-    * @param _amount the amount of bonds to be transferred
-    * @param _data additional information provided by the token holder
+    * @notice Rreturns the coupon currency, which is represented by the contract address of the token used to pay coupons. It can be the same as the one used for the principal
```
Comment (SamWilsn) [inline ERCS/erc-7092.md:236]:
```suggestion
    * @notice Returns the coupon currency, which is represented by the contract address of the token used to pay coupons. It can be the same as the one used for the principal
```

**Review by SamWilsn — APPROVED** [review]
